network/framework/src/application/storage.rs

```diff
@@ -39,6 +49,8 @@ pub struct PeersAndMetadata {
     //
     // TODO: should we remove this when generational versioning is supported?
     cached_peers_and_metadata: Arc<ArcSwap<HashMap<NetworkId, HashMap<PeerId, PeerMetadata>>>>,
+
+    subscribers: RwLock<Vec<tokio::sync::mpsc::Sender<ConnectionNotification>>>,
```

Comment:
It looks like `read()` is never called, only `write()`. Should this be changed to a mutex?

Reply:
Yup. Plain old Mutex is probably a little more efficient? I'll fix that.